### PR TITLE
doc: Add FILTER to aggregate functions

### PR DESCRIPTION
**Pull request by mkysel**
Explain the `filter` modifier for aggregations.

### CODE REVIEW

File: doc/user/content/sql/functions/filters.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+---
+title: "Aggregate Function Filters"
+description: "If FILTER is specified, then only the input rows for which the filter_clause evaluates to true are fed to the aggregate function; other rows are discarded"
+menu:
+main:
+parent: 'sql-functions'
+---
+
+If `FILTER` is specified, then only the input rows for which the `filter_clause` evaluates to true are fed to the [aggregate function](../#aggregate-func); other rows are discarded
```
Comment (umanwizard) [inline doc/user/content/sql/functions/filters.md:9]: This sentence should end with a period.

File: doc/user/content/sql/functions/filters.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+## Syntax
+
+{{< diagram "aggregate-with-filter.svg" >}}
+
+## Details
+
```
Comment (umanwizard) [inline doc/user/content/sql/functions/filters.md:16]: If this section is empty, the header can be removed.

File: doc/user/sql-grammar/sql-grammar.bnf [hunk 3]
```diff
@@ -9,7 +10,7 @@ alter_index ::=
     | 'RESET' '(' field ( ',' field )* ')'
   )
 array_agg ::=
-  'array_agg' '(' values  ( 'ORDER' 'BY' col_ref ( 'ASC' | 'DESC' )? ( ',' col_ref ( 'ASC' | 'DESC' )? )* )? ')'
+  'array_agg' '(' values  ( 'ORDER' 'BY' col_ref ( 'ASC' | 'DESC' )? ( ',' col_ref ( 'ASC' | 'DESC' )? )* )? ')' ('FILTER' '(' 'WHERE' filter_clause ')')?
```
Comment (umanwizard) [inline doc/user/sql-grammar/sql-grammar.bnf:13]: Do we describe the syntax of `filter_clause` anywhere?

Comment (mkysel) [reply doc/user/sql-grammar/sql-grammar.bnf:13]: we don't. PG does not go into any details on it, so I followed their example.

File: doc/user/content/sql/functions/filters.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
+---
+title: "Aggregate Function Filters"
+description: "If FILTER is specified, then only the input rows for which the filter_clause evaluates to true are fed to the aggregate function; other rows are discarded"
+menu:
```
Comment (msudberg) [inline doc/user/content/sql/functions/filters.md:4]: Can we boil the description down to:

> Use `FILTER` to specify which rows are sent to an aggregate function.

File: doc/user/content/sql/functions/filters.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+main:
```
Comment (msudberg) [inline doc/user/content/sql/functions/filters.md:5]: I don't think it makes a difference for processing, but to make the YAML prolog more human-readable, we usually indent main from menu and parent from main.

**Review by umanwizard — APPROVED** [review]

File: doc/user/content/sql/functions/filters.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+parent: 'sql-functions'
+---
+
+If `FILTER` is specified, then only the input rows for which the `filter_clause` evaluates to true are fed to the [aggregate function](../#aggregate-func); other rows are discarded.
```
Comment (msudberg) [inline doc/user/content/sql/functions/filters.md:9]: You can use a `FILTER` clause on an aggregate function to specify which rows are sent to an aggregate function. Rows for which the `filter_clause` evaluates to false are discarded.

**Review by msudberg — CHANGES_REQUESTED** [review]
Thanks for handling this!

Can you add an example with FILTER to the Examples sections of each of the individual aggregate functions?

**Comment by mkysel** [comment]
> Can you add an example with FILTER to the Examples sections of each of the individual aggregate functions?

It feels cluttered to me. The main function page is already very dense.

**Review by msudberg — APPROVED** [review]
lgtm!

**Pull request merged**